[Backport release-25.11] jetbrains: pycharm/idea unified builds; deprecate writerside & aqua

theCapypara opened the pull request:
Backport for #466331

The automatic backport just failed due to some changes in the maintainer lists not being backported; I backported the maintainer lists from 25.11 in a seperate commit.

theCapypara committed:
jetbrains: pycharm/idea unified products & oss builds; deprecate writerside & aqua

(cherry picked from commit 2843a9ae458e2563041ffbdaa601458a95b0b0c4)

MattSturgeon commented:
> I just used the maintainer lists from 25.11 on the new entries for this PR.

I'd prefer to consider "syncing maintainers with master" to be a [separate logical chunk](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions), so a separate commit. Assuming we can't just backport whichever commits changed the maintainer lists in the first place.

theCapypara committed:
jetbrains.idea*: backport maintainer lists from master

Contains maintainer changes from these commits (ides.json ONLY):

- 3cffb9cc70a4c54851f3765d2890f453b47ac488
- 8a64ca6114ffb5c723344f2415268aa07cfd5b25
- c0ec79be1d73c97dfdb980ad36a82951292e3ee6

Not-cherry-picked-because: Too many changes to maintainer list that can't be easily cherry picked.

theCapypara commented:
Sure, how about this?

Edit: Actually I'm not sure what I did now is allowed?

> Ensure the commits exist on the master branch. In the case of squashed or rebased merges, the commit hash will change and the new commits can be found in the merge message at the bottom of the master pull request.

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#manually-backporting-changes

MattSturgeon reviewed (approved):
Perfect, thanks.

Yeah, the point of the bot's review is to:

1. Lint any issues with cherry-picks, i.e. missing the commit hash reference.
2. Highlight any difference with the commit being cherry-picked, to aid manual review.
3. Highlight any commits that _aren't_ cherry-picked, so that they aren't glossed over in review.

This means reviewers can have confidence in commits the bot doesn't highlight, because they are clean cherry-picks of something already merged into master.

The things the bot does highlight aren't necessarily problems; they're just things that need manual review.
